experimental/web/sample_dynamic/main.c

```diff
@@ -7,23 +7,64 @@
 #include <stdint.h>
 #include <stdio.h>
 
+#include "iree/base/api.h"
+#include "iree/hal/api.h"
+#include "iree/modules/hal/module.h"
 #include "iree/runtime/api.h"
 #include "iree/vm/bytecode_module.h"
 
 //===----------------------------------------------------------------------===//
 // Public API
 //===----------------------------------------------------------------------===//
 
-int load_program(uint8_t* vmfb_data, size_t length);
+// Opaque state for the sample, shared between multiple loaded programs.
+typedef struct iree_sample_state_t iree_sample_state_t;
+
+// Initializes the sample and returns its state.
+iree_sample_state_t* setup_sample();
```

Comment:
`static` all these (always do for object-local symbols - it really helps the compiler inline/strip/etc)

Reply:
The public API functions should be exported. If I add `static` to them then the compiler strips them, leading to `emcc: error: undefined exported symbol: "_cleanup_sample" [-Wundefined] [-Werror]`. I tried the `EMSCRIPTEN_KEEPALIVE` attribute (`__attribute__((used))`) but still got that error.

https://emscripten.org/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.html#calling-compiled-c-functions-from-javascript-using-ccall-cwrap